OAI plugin: Improve error handling for missing configuration

stweil opened the pull request:
This replaces my previous pull request:

An OAI request with verb=Identity silently failed when configuration data
was missing.

Change this and show default values for admin e-mail and repository name.
Reduce also the severity of the logging from "error" to "notice".

The earliest datestamp does not depend on the configuration.
It is now calculated even when the configuration is incomplete.

Signed-off-by: Stefan Weil sw@weilnetz.de

albig commented:
Thank you for the new pull request. @sebastian-meyer will have a look at it after his holiday. I'm not so familiar with OAI yet.

sebastian-meyer commented:
That looks better! Thank you!

But maybe we should use the TYPO3 admin email as default email address (or none if that isn't set either) and something like "Goobi.Presentation OAI-PMH interface" as default repository name?

stweil commented:
I'm not sure about the e-mail address. The OAI interface is a public interface, and the e-mail address will be visible for anybody. Using an address at example.org won't harm because example.org was invented for that purpose. Using the TYPO3 admin e-mail might be an unwanted side effect which violates privacy.

The default repository name is uncritical, any name is possible there. What about  "Goobi.Presentation OAI-PMH interface (incomplete configuration)"? I'd like to show to users who test it manually (hopefully the people who did the installation) that there remains some tasks to be done.

sebastian-meyer commented:
I agree with both.

stweil committed:
OAI plugin: Improve error handling for missing configuration

An OAI request with verb=Identity silently failed when configuration data
was missing.

Change this and show default values for admin e-mail and repository name.
Reduce also the severity of the logging from "error" to "notice".

The earliest datestamp does not depend on the configuration.
It is now calculated even when the configuration is incomplete.

The new code does not pass the unneeded character encoding parameter to
htmlspecialchars.

Signed-off-by: Stefan Weil <sw@weilnetz.de>

stweil commented:
The default repository name is now  "Goobi.Presentation OAI-PMH interface (incomplete configuration)".

sebastian-meyer commented:
Thanks!
